rdt/hyper_transformer.py

```diff
@@ -155,14 +142,22 @@ def fit(self, data):
             data (pandas.DataFrame):
                 Data to fit the transformers to.
         """
-        if self.transformers is not None:
-            self._transformers = load_transformers(self.transformers)
-        else:
-            self._transformers = self._analyze(data)
-
-        for column_name, transformer in self._transformers.items():
-            column = data[column_name]
-            transformer.fit(column)
+        self._input_columns = data.columns
+        self._update_field_types(data)
+        fitted_fields = set()
+
+        for field in self.field_transformers:
+            if self._field_in_data(field, data):
+                self._fit_field_transformer(data, field, self.field_transformers[field])
+                fitted_fields.add(field)
+        for (field, data_type) in self.field_types.items():
```

Comment:
We could add some blank lines where the indentation is reduced, like above this line, or in line 160.